code/datums/components/harvester.dm

```diff
@@ -259,6 +248,25 @@
 	if(loadup_on_attack)
 		INVOKE_ASYNC(src, PROC_REF(activate_blade_async), source, user)
 
+///Handles behavior when attacking a mob with bicaridine
+/datum/component/harvester/proc/attack_bicaridine(datum/source, mob/living/target, mob/living/user, obj/item/weapon)
+	if(isxeno(target)) //Self-heal on xeno strike
+		new /obj/effect/temp_visual/telekinesis(get_turf(user))
+		target.apply_damage(weapon.force*0.6, BRUTE, user.zone_selected)
+		user.adjustStaminaLoss(-30)
+		user.heal_overall_damage(5, 0, updating_health = TRUE)
+		return
+
+	to_chat(user, span_rose("You prepare to stab <b>[target != user ? "[target]" : "yourself"]</b>!"))
+	new /obj/effect/temp_visual/telekinesis(get_turf(target))
+
+	if(do_after(user, 2 SECONDS, TRUE, target, BUSY_ICON_DANGER)) //Channeled heal on human strike
+		var/skill_heal_amt = user.skills.getRating(SKILL_MEDICAL) * 5
+		target.heal_overall_damage(10 + skill_heal_amt, 0, updating_health = TRUE) //5u of Bica will normally heal 25 damage. Medics get this full amount
+	else
+		target.adjustStaminaLoss(-30)
```

Comment:
I know you didn't do this part, but why the fuck does it heal stam if you SKIP the do_after?

Reply:
The stamina heal is attached to the lesser heal.
The do_after is only for the greater heal.

Reply:
yes, so you can just interrupt the do_after for instant stam heal? fucking vali man

Reply:
Yes